File: Programacao_Concorrente/java/src/imd/ufrn/br/bilheteria/Ticket.java
package imd.ufrn.br.bilheteria;

public class Ticket {
	private int value;
	private final int maxTickets;
	private int soldTickets;
	
	public Ticket(int n, int max) {
		value = n;
		maxTickets = max;
		soldTickets = 0;
	}
	
	public int getTicket()
	{
		if(soldTickets >= maxTickets) { return -1;}
		soldTickets++;
		return value++;
	}
	
	public int getAvailableTickets() {
		return maxTickets - soldTickets;
	}
}

File: Programacao_Concorrente/c#/Bilheteria/src/Ticket.cs
namespace Bilheteria;

public class Ticket(int n, int max)
{
    private int value = n;
    private readonly int max_tickets = max;
    private int sold_tickets = 0;

    public int GetTicket(){
        if(sold_tickets > max_tickets){
            return -1;
        }
        
        sold_tickets++;
        return value++;
    }

    public int GetAvailableTickets(){
        return max_tickets-sold_tickets;
    }
}